scanreport.sh

```diff
@@ -32,17 +35,19 @@ fi
 
 if [[ ! -z $ip ]]; then # search by IP
 	r=$(grep -w $ip "$db" | grep -v ^# | sed 's/Ports: /\'$'\n/g' |  tr '/' '\t' | tr ',' '\n' | sed 's/^ //g' | grep -v "Status: Up" | sed 's/Host:/\\033[0;32mHost:\\033[0;39m/g' | sed 's/Ignored State.*$//')
-	echo -e "$r"
 
 elif [[ ! -z $port ]]; then # search by port number
 	r=$(grep -w -E -e "($port)\/open" "$db" | grep -v ^# | sed 's/Ports: /\'$'\n/g' |  tr '/' '\t' | tr ',' '\n' | sed 's/^ //g' | grep -v "Status: Up" | grep -E -e "Host: " -e "^(${port})" | sed 's/Host:/\\033[0;32mHost:\\033[0;39m/g' | sed 's/Ignored State.*$//')
-	echo -e "$r"
 
 elif [[ ! -z $service ]]; then # search by service name
 	r=$(grep -w -E -i -e "($service)" "$db" | grep -v ^# | sed 's/Ports: /\'$'\n/g' |  tr '/' '\t' | tr ',' '\n' | sed 's/^ //g' | grep -v "Status: Up" | grep -i -E -e "Host: " -e "(${service})" | sed 's/Host:/\\033[0;32mHost:\\033[0;39m/g' | sed 's/Ignored State.*$//')
-	echo -e "$r"
 
 else
 	r=$(cat "$db" | grep -v ^# | sed 's/Ports: /\'$'\n/g' | tr '/' '\t' | tr ',' '\n' | sed 's/^ //g' | grep -v "Status: Up" | sed 's/Host:/\\033[0;32mHost:\\033[0;39m/g' | sed 's/Ignored State.*$//')
-	echo -e "$r"
 fi
+
+if [[ ! -z $addressonly ]]; then # output only IPs/hostnames
+	echo -e "$r" | grep "Host:" | awk {'print $2'}
```

Comment:
If OPTARG is used above, then this should work no?

`echo -e "$r" | grep "Host:" | $addressonly`